### DISCUSSION TITLE
Getting "WEBP compression support is not configured" trying to make pyramid tiff

### DISCUSSION THREAD

**Discussion by nator** [libvips/libvips#2054] [Q&A]
I'm trying to use webp compression when writing a pyramid tiff, both for file size and as a way to keep alpha channel. I feel like I'm missing an obvious step somewhere, but I've read all the docs I can find and can't see what I'm missing... I'm currently using cli since sharp doesn't expose webp as a tiff output. Would I have better luck jumping to pyvips maybe?

Basic command is
`vips tiffsave Test_Source_Alpha.tif test_webp.tif --pyramid --compression webp`

Output is
```
vips temp-1: 2550 x 3300 pixels, 16 threads, 2550 x 16 tiles, 384 lines in buffer
vips temp-1: done in 0.0203s          
TIFFSetField: test_webp.tif: Unknown pseudo-tag 65568
TIFFSetField: test_webp.tif: Unknown pseudo-tag 65569
TIFFSetField: /var/folders/7y/yfrw7n3j6lz8zmblgmsbcb1hzvsv3h/T/vips-0-1102629798.tif: Unknown pseudo-tag 65568
TIFFSetField: /var/folders/7y/yfrw7n3j6lz8zmblgmsbcb1hzvsv3h/T/vips-0-1102629798.tif: Unknown pseudo-tag 65569
TIFFSetField: /var/folders/7y/yfrw7n3j6lz8zmblgmsbcb1hzvsv3h/T/vips-1-2430140760.tif: Unknown pseudo-tag 65568
TIFFSetField: /var/folders/7y/yfrw7n3j6lz8zmblgmsbcb1hzvsv3h/T/vips-1-2430140760.tif: Unknown pseudo-tag 65569
TIFFSetField: /var/folders/7y/yfrw7n3j6lz8zmblgmsbcb1hzvsv3h/T/vips-2-4099540810.tif: Unknown pseudo-tag 65568
TIFFSetField: /var/folders/7y/yfrw7n3j6lz8zmblgmsbcb1hzvsv3h/T/vips-2-4099540810.tif: Unknown pseudo-tag 65569
TIFFSetField: /var/folders/7y/yfrw7n3j6lz8zmblgmsbcb1hzvsv3h/T/vips-3-666550103.tif: Unknown pseudo-tag 65568
TIFFSetField: /var/folders/7y/yfrw7n3j6lz8zmblgmsbcb1hzvsv3h/T/vips-3-666550103.tif: Unknown pseudo-tag 65569
TIFFSetField: /var/folders/7y/yfrw7n3j6lz8zmblgmsbcb1hzvsv3h/T/vips-4-490388930.tif: Unknown pseudo-tag 65568
TIFFSetField: /var/folders/7y/yfrw7n3j6lz8zmblgmsbcb1hzvsv3h/T/vips-4-490388930.tif: Unknown pseudo-tag 65569
test_webp.tif: WEBP compression support is not configured
vips2tiff: TIFF write tile failed
wbuffer_write: write failed
unix error: Unknown error: -1
```

I installed vips using homebrew on a mac: `vips-8.10.1-Fri Sep  4 11:40:49 UTC 2020` and `vips webpsave` and `vipsthumbnail` can produce good webp images.

I'm not attaching source image since this seems to happen with _any_ source, even simple jpegs. What am I missing to configure webp compression? Thanks for any help!

**Comment by jcupitt** [comment]
Hello @nator,

It's working for me on Ubuntu 20.04:

```
vips copy k2.jpg k2.tif[compression=webp]
```

I would guess that the homebrew libtiff formula is not enabling webp. I think I would examine that.

**Comment by jcupitt** [comment]
Sorry, that was too brief. Webp in tiff needs a libtiff that's been built against libwebp, and that seems to be what's failing.  The message:

```
test_webp.tif: WEBP compression support is not configured
```

is being generated by libtiff, not by libvips.

**Comment by nator** [comment]
Ah, brilliant - I was searching everywhere for that error string and coming up empty. I'll see if I can get homebrew to work with a custom libtiff or if needed just build everything from scratch.

I really really appreciate both the tool and your amazing responsiveness. :)

**Comment by jcupitt** [comment]
I think you can probably just edit the `tiff.rb` formula in `/usr/local` and rebuild it, it should be simple.

**Comment by nator** [comment]
Me again, sorry. I'm trying now to get this working on Windows, for various terrible reasons. :/ I've followed the steps in the [windows build repo](https://github.com/libvips/build-win64) with the following changes in `vips.modules`, but it doesn't seem to have worked:
```
  <autotools id="tiff"
    autogen-sh="configure"
    autogenargs="--without-x --disable-cxx --disable-lzma --enable-webp">
    <branch
      repo="tiff"
      module="tiff-${version}.tar.gz"
      version="4.1.0"/>
    <dependencies>
      <dep package="zlib"/>
      <dep package="webp"/>
      <dep package="libjpeg-turbo"/>
    </dependencies>
  </autotools>
```
Is that close at least? I didn't notice any errors while building and it did produce good binaries, just no webp for libtiff.

I finally installed MS Visual Studio and used cmake to build libtiff with webp (gathered jpeg, zlib, and webp libraries for building and linking) and if I rename that dll and replace libtiff-5.dll in vips/bin, I have success.

Very low priority since I have a workaround, but can you spot what I'm doing wrong with the official build steps? For most people I think that would be easier than the cmake hoops I had to jump through.

Thanks!
